### PR TITLE
Initial commit of NestJS tfb configuration - included tests - json, p…

### PR DESCRIPTION
**Pull request by jenriquez-techempower**
…laintext, updates, multiple, single

### CODE REVIEW

File: frameworks/TypeScript/nest/src/bench/bench.service.ts [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+            const worldArr = [];
+            const total = this.sanitizeQueries(totalQueries);
+            for (let i = 0; i < total; i++) {
+                let worldToUpdate = await this.getOne();
+                worldToUpdate.randomnumber = Math.floor(Math.random() * 10000) + 1;
+                worldToUpdate = await this.worldRepository.save(worldToUpdate);
+                worldArr.push(worldToUpdate);
+            }
+
+            if (worldArr.length === total) {
```
Comment (nbrady-techempower) [inline frameworks/TypeScript/nest/src/bench/bench.service.ts]: Should you remove this condition? What happens if the total isn't reached? Does the server respond? Does it error with a rejected promise? If you remove the condition and the total is wrong, the toolset will complain about the number of results returned versus the number of results expected.

File: frameworks/TypeScript/nest/tslint.json [hunk 2]
```diff
@@ -0,0 +1,55 @@
+{
```
Comment (nbrady-techempower) [inline frameworks/TypeScript/nest/tslint.json]: Linting stuff can also be removed, but kept locally if you want it.

File: frameworks/TypeScript/nest/src/bench/bench.service.ts (hunk 1 above)
Comment (jenriquez-techempower) [reply frameworks/TypeScript/nest/src/bench/bench.service.ts]: Removed. It was unnecessary.

File: frameworks/TypeScript/nest/src/app.service.ts [hunk 3]
```diff
@@ -0,0 +1,8 @@
+import { Injectable } from '@nestjs/common';
```
Comment (jcoleman-techempower) [inline frameworks/TypeScript/nest/src/app.service.ts]: We may be able to do away with all the boilerplate code and just leave the `BenchController`

File: frameworks/TypeScript/nest/src/bench/bench.controller.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    async getMultipleRandomWorlds(@Query('queries') query){
+
+        return await this.benchService.getMultiple(query);
+    }
+
+    @Get('/updates')
+    @Header('Server', 'NestJS')
+    async updateRandomWorlds(@Query('queries') query){
+
+        return await this.benchService.updateMultiple(query);
```
Comment (jcoleman-techempower) [inline frameworks/TypeScript/nest/src/bench/bench.controller.ts]: We don't need to await this line. Only the line where this function is called as `async updateRandomWorlds` will already return the Promise.

Comment (jcoleman-techempower) [reply frameworks/TypeScript/nest/src/bench/bench.controller.ts]: You can apply the same logic to the other methods in this file.

File: frameworks/TypeScript/nest/src/bench/bench.service.ts [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+
+@Injectable()
+export class BenchService {
+    constructor(
+        @InjectRepository(WorldEntity)
+        private worldRepository: Repository<WorldEntity>){}
+
+    async getOne(){
+
+        return await this.worldRepository.findOne({ where: {id: Math.floor(Math.random() * 10000) + 1}});
```
Comment (jcoleman-techempower) [inline frameworks/TypeScript/nest/src/bench/bench.service.ts]: No need to `await` here, either.

**Pull request merged**